src/Settings.js

```diff
@@ -24,6 +24,7 @@ class Settings {
         this.errorColor = utils.deflt(options.errorColor, "#cc0000");
         this.macros = options.macros || {};
         this.colorIsTextColor = utils.deflt(options.colorIsTextColor, false);
+        this.maxSize = utils.deflt(options.maxSize, 0);
```

Comment:
Maybe this could be `Math.max(0, utils.deflt(options.maxSize, Infinity))`.  I'm not sure if anything bad will happen from having a negative `maxSize`, but it doesn't make sense so we may as well guard against it.